### DISCUSSION TITLE
classes/class_animatablebody3d

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#230] [User-contributed notes]
# classes/class_animatablebody3d

Inherits: StaticBody3D< PhysicsBody3D< CollisionObject3D< Node3D< Node< Object A 3D physics body that can't be moved by external forces. When moved manually, it affects other bodies in its path. De...

https://docs.godotengine.org/en/stable/classes/class_animatablebody3d.html

**Comment by shegedep** [comment]
AnimatableBody performs the job that StaticBody did in 3.x when the property “infer_velocity” was enabled.

Essentially the AnimatableBody will automatically calculate its linear and angular velocity no matter how you reposition it. This is useful for collision response, such as a CharacterBody treating it as a moving platform.

NOTE: If you are *not* moving the body via Animations, you almost certainly want to leave “sync_to_physics” disabled. A more suitable name for this property could be “sync_animation_frame_to_physics”.

**Comment by superhighlevel** [comment]
AnimatableBody3D will ***not*** move with parent with `sync_to_physics = true`, if you don't use with AnimationPlayer set `sync_to_physics = false`

**Reply by Chestnut45** [reply]
It can still be useful to disable ```sync_to_physics``` even if you are using AnimationPlayer to animate a node's position, if you want the animation to play locally. (i.e. a melee weapon scene that plays a spinning animation in local space while still attached to a hand bone of the player model for its parent transform works as expected with this disabled)

Probably the property's description should include that it ignores parent transforms during animation playback. (in 4.5 stable using jolt at least, it seems to start respecting the parent transform again after an animation ends, but from a new offset if the parent node has moved during playback)

**Reply by Nodragem** [reply]
this is such a mess! I'm trying to open a door using AnimatableBody3D. The AnimatableBody is a child of a larger object and it is animated through AnimationPlayer, which leads to animation not playing. 

I had to unable `sync_to_physics` for the animation to play, but I am not sure what I am loosing doing that?
Furthermore, note that everything was working well in Editor!

**Reply by waltribeiro** [reply]
After 2 hours of debugging, this comment thread fixed it for me. The checkbox "Sync To Physics" is a bit confusing, especially since what I saw in my animation player was not what was happening at runtime. I couldn't figure it out. Thanks everyone.

**Reply by YangQuuQgnaY** [reply]
I also encountered a similar situation.The only solution is to close this.I was using a Pivot to rotate a platform.In Editor, CollisionShape3D rotating well.But in game it was static.I'm really confused about "Sync To Physic"!
